### PR TITLE
fix: warning on RN v0.65+

### PR DESCRIPTION
**Pull request by enagorny**
## Summary

`removeEventListener` is deprecated for a several years https://reactnative.dev/docs/0.65/appstate#removeeventlistener

## Changelog

[General] [Fix] - Warning on react-native 0.65+

### CODE REVIEW

**Comment by kevindice** [comment]
This addresses #5371

**Comment by kevindice** [comment]
Hi @rayan1810, I've tested this locally and it solves problem for me with React `0.70.1`.  Could we please get this in the next release?  Thanks!

File: src/core/color-mode/hooks.tsx [hunk 1]
```diff
@@ -31,8 +31,18 @@ export const useAppState = () => {
     () => ({
       getCurrentValue: () => AppState.currentState,
       subscribe: (callback: () => void) => {
-        AppState.addEventListener('change', callback);
-        return () => AppState.removeEventListener('change', callback);
+        const subsription = AppState.addEventListener('change', callback);
+        // return () => AppState.removeEventListener('change', callback);
```
Comment (kevindice) [inline src/core/color-mode/hooks.tsx:35]: Remove dead code.

```suggestion
```

**Comment by austinrbrown01** [comment]
+1 to please push in next update

**Pull request merged**